### PR TITLE
GDBus

### PR DESCRIPTION
**Pull request by GeorgesStavracas**
Switch `libobs/util/platform-nix-dbus.c` to GDBus, and do some smaller cleanups while at it.

### Motivation and Context

GDBus is more and better maintained than libdbus these days. In the future, a potential Wayland-compatible capture plugin will need to interact with D-Bus in a way that's way too complicated for libdbus, and it won't be nice to have both libraries talking to the D-Bus socket.

### How Has This Been Tested?

On Linux, start screencasting and the session shouldn't inhibit.

### Types of changes

 - Code cleanup (non-breaking change which makes code smaller or more readable)

### Checklist:

- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

### CODE REVIEW

**Comment by GeorgesStavracas** [comment]
Hm, the "Clang Format Check" failure seems unrelated - it failed to install clang.

**Comment by GeorgesStavracas** [comment]
This needs to be tested on:

 * Mate
 * Any super old system that uses `org.gnome.SessionManager`
 * Any system that uses `org.freedesktop.PowerManager.Inhibitor`

**Comment by flexiondotorg** [comment]
> This needs to be tested on:
> 
> * Mate

I can do that.

File: cmake/Modules/FindGio.cmake [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  # in the find_path() and find_library() calls
+  find_package(PkgConfig)
+  pkg_check_modules(PC_GIO gio-2.0)
+
+  set(GIO_DEFINITIONS ${PC_GIO_CFLAGS})
+
+  find_path(GIO_INCLUDE_DIRS gio.h PATHS ${PC_GIO_INCLUDEDIR} ${PC_GIO_INCLUDE_DIRS} PATH_SUFFIXES glib-2.0/gio/)
+  find_library(GIO_LIBRARIES NAMES gio-2.0 libgio-2.0 PATHS ${PC_GIO_LIBDIR} ${PC_GIO_LIBRARY_DIRS})
+
+  if(GIO_INCLUDE_DIRS AND GIO_LIBRARIES)
```
Comment (kkartaltepe) [inline cmake/Modules/FindGio.cmake:21]: Use find_package_handle_standard_args if you are writing a new find module. Perhaps there is already one being maintained in upstream cmake already?

Comment (GeorgesStavracas) [reply cmake/Modules/FindGio.cmake:21]: Hm, what's "upstream" here? As far as I can see, GLib itself doesn't ship any CMake modules.

Comment (GeorgesStavracas) [reply cmake/Modules/FindGio.cmake]: In any case, switched to `find_package_handle_standard_args`

**Comment by GeorgesStavracas** [comment]
It seems the Mate and GNOME Session D-Bus calls are working. I will assume that `org.freedesktop.PowerManager.Inhibitor` will as well, since it has the exact same signature of Mate's and GNOME Session's.

File: cmake/Modules/FindGio.cmake [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+  find_path(GIO_INCLUDE_DIRS gio.h PATHS ${PC_GIO_INCLUDEDIR} ${PC_GIO_INCLUDE_DIRS} PATH_SUFFIXES glib-2.0/gio/)
+  find_library(GIO_LIBRARIES NAMES gio-2.0 libgio-2.0 PATHS ${PC_GIO_LIBDIR} ${PC_GIO_LIBRARY_DIRS})
+  mark_as_advanced(GIO_INCLUDE_DIRS GIO_LIBRARIES)
+
+  include(FindPackageHandleStandardArgs)
+  find_package_handle_standard_args(Gio REQUIRED_VARS GIO_LIBRARIES GIO_INCLUDE_DIRS)
+
+  if(GIO_FOUND)
+    set(HAVE_DBUS "1")
```
Comment (kkartaltepe) [inline cmake/Modules/FindGio.cmake:26]: wlll `set(HAVE_DBUS "${GIO_FOUND}")` work?

File: cmake/Modules/FindGio.cmake [hunk 3]
```diff
@@ -0,0 +1,30 @@
+# - Try to find Gio
+# Once done this will define
+#
+#  GIO_FOUND - system has Gio
+#  GIO_INCLUDE_DIRS - the Gio include directory
+#  GIO_LIBRARIES - the libraries needed to use Gio
+#  GIO_DEFINITIONS - Compiler switches required for using Gio
+
+if(NOT WIN32)
```
Comment (kkartaltepe) [inline cmake/Modules/FindGio.cmake:9]: windows checks are not needed.

File: cmake/Modules/FindGio.cmake (hunk 2 above)
Comment (GeorgesStavracas) [reply cmake/Modules/FindGio.cmake]: It looks like it doesn't work transparently. If you're feeling strong about it, I can add a new commit changing how `HAVE_DBUS` is defined throughout CMake.

**Comment by ColdMoonNight** [comment]
Do you need a tester on KDE?

**Comment by GeorgesStavracas** [comment]
> Do you need a tester on KDE Plasma?

That would be great! Absolutely no change in behaviour is expected here - it's should work exactly equal on GNOME, KDE, XFCE, etc. (Of course unless this PR introduces any regression!)

File: cmake/Modules/FindGio.cmake (hunk 2 above)
Comment (kkartaltepe) [reply cmake/Modules/FindGio.cmake]: no i just wasnt able to test myself, this is fine.

File: cmake/Modules/FindGio.cmake [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+#  GIO_DEFINITIONS - Compiler switches required for using Gio
+
+# Use pkg-config to get the directories and then use these values
+# in the find_path() and find_library() calls
+find_package(PkgConfig)
+pkg_check_modules(PC_GIO gio-2.0)
+
+set(GIO_DEFINITIONS ${PC_GIO_CFLAGS})
+
+find_path(GIO_INCLUDE_DIRS gio.h PATHS ${PC_GIO_INCLUDEDIR} ${PC_GIO_INCLUDE_DIRS} PATH_SUFFIXES glib-2.0/gio/)
```
Comment (kkartaltepe) [inline cmake/Modules/FindGio.cmake:16]: Internally finds should be the old style FOO_INCLUDE_DIR and FOO_LIBRARY, then the outputs created by find_package_handle_standard_args will produce the standard FOO_INCLUDE_DIRS and FOO_LIBRARIES... Cmake has a lot of weird traditions.

Comment (kkartaltepe) [reply cmake/Modules/FindGio.cmake:16]: https://cmake.org/cmake/help/latest/module/FindPackageHandleStandardArgs.html sort of tangentially mentioned here.

**Review by kkartaltepe — APPROVED** [review]

File: cmake/Modules/FindGio.cmake (hunk 4 above)
Comment (kkartaltepe) [reply cmake/Modules/FindGio.cmake:16]: err im wrong you just set them manually in the canonical example. https://cmake.org/cmake/help/latest/manual/cmake-developer.7.html#a-sample-find-module

Comment (GeorgesStavracas) [reply cmake/Modules/FindGio.cmake:16]: I see. Well, I usually look at how KDE folks use CMake, and it seems to be a common practice to pass `LIBRARIES` / `DEFINITIONS` / `DIRS` directly to `find_*()`. [Example](https://github.com/KDE/krfb/blob/master/cmake/modules/FindPipeWire.cmake).

**Pull request merged**